app/models/product_import/spreadsheet_data.rb

```diff
@@ -1,3 +1,10 @@
+# This class encapsulates a number of "indexes" used during product import. These contain hashes
+# of information that need to be accessed at various stages of the import, and are built in order
+# to minimise the number of queries that take place. So for instance, if a spreadsheet has 4000
+# products for 5 different enterprises and we need to check the enterprise permissions for each
+# product during validation, we have a small index for that data that gets built at the beginning
+# so we don't have to make 4000 queries.
+
 module ProductImport
   class SpreadsheetData
```

Comment:
According to description, I'd call this class ProductImportSupportIndexes.
SpreedsheetData sounds like all the spreadsheet data is available in this class.